generate_figures/utilities.py

```diff
@@ -0,0 +1,63 @@
+import pandas as pd
+import wandb
+
+api = wandb.Api()
+
+def get_plot_params(figsize, fontsize, ticksize):
+    params = {
+        "text.usetex": True,
+        "font.family": "serif",
+        "figure.figsize": figsize,
+        "figure.dpi": 80,
+        "figure.edgecolor": "k",
+        "font.size": fontsize,
+        "axes.labelsize": fontsize,
+        "axes.titlesize": fontsize,
+        "xtick.labelsize": ticksize,
+        "ytick.labelsize": ticksize,
+    }
+    return(params)
+
+
+def get_results_by_tag(project, tag, cols_config = [], test_results = False, cols_summary = []):
+    # The `cols` will always automatically ahve the run name in it, which will help with filtering to check if there was more than one run since the new datastructure coudl have multiple rows per run.
+    # if test_results = False then just collect all of the cols and cols_config and create a dataframe with them.
+    # if test_results = True then create a dataframe by pulling down the test_results artifact, addd the cols and cols_config to that dataframe (which makes repeition but that is OK for our purposes here), and then just concatenates all of those dataframes together
+   # return a single dataframe regardless.
+    runs = api.runs(project, filters={"tags": tag})
+    if test_results == False:
+        d=[]
```

Comment:
My suggestion is that you create what is the `d` in common whether you add the `test_results` or not - so you have shared code.  Then after you have the `d` you can check whether you want it to get the `test_results` and work accordingly.